lib/engine/game/g_18_usa/game.rb

```diff
@@ -630,20 +630,78 @@ def operating_round(round_num)
             G1817::Step::CashCrisis,
             G18USA::Step::ObsoleteTrain,
             G18USA::Step::Loan,
+            G18USA::Step::ScrapTrain,
             G18USA::Step::SpecialTrack,
             G18USA::Step::SpecialToken,
             G18USA::Step::SpecialBuyTrain,
             G18USA::Step::Assign,
             G18USA::Step::Track,
             G18USA::Step::DenverTrack,
             G18USA::Step::Token,
+            G18USA::Step::BuyPullman,
             G18USA::Step::Route,
             G18USA::Step::Dividend,
-            Engine::Step::DiscardTrain,
+            G18USA::Step::DiscardTrain,
             G18USA::Step::BuyTrain,
           ], round_num: round_num)
         end
 
+        def pullman_scrap_value
+          50
+        end
+
+        def scrap_info
+          "Scrap Pullman for #{format_currency(pullman_scrap_value)}"
+        end
+
+        def scrap_button_text
+          'Scrap Pullman'
+        end
+
+        def use_compact_scrap_trains_view
+          true
+        end
+
+        def use_1840_style_merger_round_scrap_trains?
+          false
+        end
+
+        def crowded_corps
+          @crowded_corps ||= corporations.select do |c|
+            trains = self.class::OBSOLETE_TRAINS_COUNT_FOR_LIMIT ? c.trains.size : c.trains.count { |t| !t.obsolete }
+            trains > train_limit(c) || c.trains.count { |t| pullman_train?(t) } > 1
+          end
+        end
+
+        # owner is the alleged corporation scrapping a pullman
+        def scrap_train_by_corporation(action, _owner)
+          entity = action.entity
+          raise GameError, "#{entity.name} cannot scrap a train now" unless entity == current_entity
+
+          train = action.train
+          raise GameError, "#{entity.name} cannot scrap a #{train.name} train" unless pullman_train?(train)
+
+          scrap_train(train)
+        end
+
+        # owner is the alleged owner of the company scrapping a pullman
+        def scrap_train_by_owner(action, _owner)
+          entity = action.entity
+          raise GameError, "#{entity.name} cannot scrap a train now" unless entity&.owner == current_entity
+
+          train = action.train
+          raise GameError, "#{entity.name} cannot scrap a #{train.name} train" unless pullman_train?(train)
+
+          scrap_train(train)
+        end
+
+        # Do error checking before calling this.
+        def scrap_train(train)
+          @bank.spend(pullman_scrap_value, train.owner)
+          @log << "#{train.owner.name} scraps a pullman for #{format_currency(pullman_scrap_value)}"
+          @depot.reclaim_train(train)
+        end
```

Comment:
These 3 methods should also be looked at to see if they fit better in the scrap train step.

Reply:
These are also put into scrap train module